generator/generate.go

```diff
@@ -56,6 +57,12 @@ func Generate(pattern string, mapping []comments.Converter, config Config) (*jen
 		}
 		interf := obj.Type().Underlying().(*types.Interface)
 
+		// first, load extend methods from the command line / global config.
+		if err := gen.parseExtend(obj.Type(), sources.Scope(), config.ExtendMethods); err != nil {
+			return nil, fmt.Errorf("Error while parsing extend methods: %s", err)
+		}
```

Comment:
No reasons, I merged the array. Except that your suggestion was not `technically` correct: 
`extendMethods := append(converter.Config.ExtendMethods, config.ExtendMethods...)` has potential side effect.

This is due to how golang's append works and I personally hate that this side affect ... but this is just my personal opinion :). If the input param (converter.Config.ExtendMethods in this case) has enough capacity, append will modify the input and add new elements to it. If not, append allocates new array and input stays as is. To prevent that, new array must be allocated to store results.

Reply:
Interesting, I didn't know that, thanks!